script.js

```diff
@@ -1,4 +1,161 @@
+var allScoresDict = {};
+var playerScores = [];
+var GAME_STARTED = false;
+var gameMode;
+
+// function that roll dice randomly
+var rollDice = function (input) {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal) + 1;
+  return randomInteger;
+};
+
+// roll a random dice and push it to the scores
+var playerRollDice = function () {
+  var counter = 0;
+  while (counter < 2) {
+    playerScores.push(rollDice());
+    counter += 1;
+  }
+  return (
+    `Welcome player ${currentPlayer}, <br> Your current dice rolls are: ` +
+    playerScores[0] +
+    " and " +
+    playerScores[1] +
+    ". <br> Choose the order of the dice by entering 1 or 2."
+  );
+};
+
+// concatenate 2 scores to form a number
+var concatScores = function (num1, num2) {
+  return Number(String(num1) + String(num2));
+};
+
+// dice ordering
+var getPlayerScore = function (playerInput) {
+  var combineScore;
+  console.log(typeof playerInput);
+
+  if (playerInput != 1 && playerInput != 2) {
+    return "Error! Please key in either 1 or 2";
+  } else if (playerInput == 1) {
+    combineScore = concatScores(playerScores[0], playerScores[1]);
+  } else if (playerInput == 2) {
+    combineScore = concatScores(playerScores[1], playerScores[0]);
+  }
+  //storing the scores of each player in a dictionary
+  allScoresDict[currentPlayer] = combineScore;
+  playerScores = [];
+  return `Player: ${currentPlayer}, your chosen number is: ${combineScore}`;
+};
+
+// normal mode of determine winner
+var determineWinner = function () {
+  if (allScoresDict[1] > allScoresDict[2]) {
+    return `Player 1 wins! <br> Player 1's score: ${allScoresDict[1]} <br> Player 2's score: ${allScoresDict[2]}`;
+  }
+  if (allScoresDict[1] == allScoresDict[2]) {
+    return `It's a draw!! <br> Player 1's score: ${allScoresDict[1]} <br> Player 2's score: ${allScoresDict[2]}`;
+  }
+  if (allScoresDict[1] < allScoresDict[2]) {
+    return `Player 2 wins! <br> Player 1's score: ${allScoresDict[1]} <br> Player 2's score: ${allScoresDict[2]}`;
+  }
+};
+
+// reverse mode of determining winner
+var reverseWinner = function () {
+  if (allScoresDict[1] < allScoresDict[2]) {
+    return `Player 1 wins! <br> Player 1's score: ${allScoresDict[1]} <br> Player 2's score: ${allScoresDict[2]}`;
+  }
+  if (allScoresDict[1] == allScoresDict[2]) {
+    return `It's a draw!! <br> Player 1's score: ${allScoresDict[1]} <br> Player 2's score: ${allScoresDict[2]}`;
+  }
+  if (allScoresDict[1] > allScoresDict[2]) {
+    return `Player 2 wins! <br> Player 1's score: ${allScoresDict[1]} <br> Player 2's score: ${allScoresDict[2]}`;
+  }
+};
+
+// reset the game back to its original state
+var gameReset = function () {
+  playerScores = [];
+  currentPlayer = 1;
+  allScoresDict = {};
+  game_state = ROLL_DICE;
+  GAME_STARTED = false;
+  return "Click submit to play again";
+};
+
+//main function
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
```

Comment:
I would suggest having as little code in your var main as possible. Would have the game logic code in a larger function for basic, comfortable and more comfortable. Then your var main can just 
```
return basicGameLogic(input)
```